aggregator/src/aggregator/aggregation_job_writer.rs

```diff
@@ -197,22 +197,41 @@ impl<const SEED_SIZE: usize, Q: CollectableQueryType, A: vdaf::Aggregator<SEED_S
             })
             .collect();
 
-        // Read all relevant batches from the datastore.
-        let mut batches: HashMap<_, _> = try_join_all(self.by_batch_identifier_index.keys().map(
-            |batch_identifier| {
-                tx.get_batch::<SEED_SIZE, Q, A>(
-                    self.task.id(),
-                    batch_identifier,
-                    aggregation_parameter,
-                )
-            },
-        ))
-        .await?
-        .into_iter()
-        .flat_map(|batch: Option<Batch<SEED_SIZE, Q, A>>| {
-            batch.map(|b| (b.batch_identifier().clone(), b))
-        })
-        .collect();
+        // Read all relevant batches & report counts from the datastore.
+        let (batches, batches_with_reports) = try_join!(
+            try_join_all(
+                self.by_batch_identifier_index
+                    .keys()
+                    .map(|batch_identifier| {
+                        tx.get_batch::<SEED_SIZE, Q, A>(
+                            self.task.id(),
+                            batch_identifier,
+                            aggregation_parameter,
+                        )
+                    })
+            ),
```

Comment:
These two `try_join_all!` calls will create one DB query for each batch identifier. Would it be more efficient to write a new SQL query that gets all batches matching a list of `batch_identifiers` in one query? If so, we should file an issue to change that separately from this change.

Reply:
This is https://github.com/divviup/janus/issues/224#issuecomment-1511897659 -- from prior discussions, it might be more performant, but we believe the performance difference to be marginal due to the use of request pipelining.